src/paymentRequest/IncomingPaymentRequestListItem.tsx

```diff
@@ -63,26 +69,58 @@ export default function IncomingPaymentRequestListItem({ id, amount, comment, re
       notificationType: NotificationBannerTypes.incoming_tx_request,
       selectedAction: NotificationBannerCTATypes.decline,
     })
-    dispatch(declinePaymentRequest(id))
+    dispatch(declinePaymentRequest(paymentRequestId))
     Logger.showMessage(t('requestDeclined'))
   }
 
   const navigateToNextScreen = () => {
-    const transactionData: TransactionDataInput = {
-      reason: comment,
-      recipient: requester,
-      amount: new BigNumber(amount),
-      currency: Currency.Dollar,
-      type: TokenTransactionType.PayRequest,
-      firebasePendingRequestUid: id,
+    const cUsdTokenInfo = stableTokens.find((token) => token?.symbol === Currency.Dollar)
+    const cEurTokenInfo = stableTokens.find((token) => token?.symbol === Currency.Euro)
+    if (!cUsdTokenInfo?.address || !cEurTokenInfo?.address) {
+      // Should never happen in producton
+      throw new Error('No token address found for cUSD')
+    }
+    // If the user has enough cUSD balance, pay with cUSD
+    // Else, try with cEUR
+    // Else, throw up an error banner
+    let transactionData: TransactionDataInput
+    const usdRequested = new BigNumber(paymentRequest.amount)
```

Comment:
is paymentRequest.amount always in usd? even if the local currency is different?

Reply:
Yep thats how it was implemented :)

Reply:
poorly named back then to just be 'amount'